File: AppMovil/lib/controllers/notificaciones_controller.dart
import 'package:NotiPush/models/notificacion.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class NotificacionesController {
  static final String _baseUri = 'http://192.168.1.110:3000';

  static Future<dynamic> getNotificaciones(num usuarioId) async {
    List<Notificacion> notificaciones = new List();

    final request =
        await http.get('$_baseUri/notificaciones?usuario_id=$usuarioId');
    final response = jsonDecode(request.body);

    if (response['estatus']) {
      List datosNotificaciones = response['notificaciones'];

      datosNotificaciones.forEach((element) {
        if (element['titulo'] != null) {
          Notificacion notificacion = Notificacion.fromJson(element);

          notificaciones.add(notificacion);
        }
      });
    }

    return notificaciones;
  }

  static Future<dynamic> getNotificacion(num id) async {
    print('hola');
    final request = await http.get('$_baseUri/notificaciones/single?id=$id');
    final response = jsonDecode(request.body);
    print(response);
    if (response['estatus']) {
      Notificacion not = Notificacion.fromJson(response['notificacion']);

      return not;
    }

    return response['mensaje'];
  }
}

File: SistemaWeb/app/Http/Controllers/NotificacionesController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\Http;
use Illuminate\Support\Facades\Auth;

use App\Models\Notificacion;
use Carbon\Carbon;
use Illuminate\Support\Facades\Redirect;

class NotificacionesController extends Controller
{
    //
    public function index() {
        $notificaciones = $this->getNotificaciones();
        $grupos = $this->getGrupos();
        
        $array = [
            'notificaciones' => $notificaciones,
            'grupos' => $grupos,
        ];

        return view('notifications', $array);
    }
    
    public function getNotificaciones()
    {
        $notificaciones = [];
        $peticion = Http::get('http://localhost:3000/notificaciones/', [
            'usuario_id' => Auth::user()->id,
        ]);

        $respuesta = json_decode($peticion->getBody()->getContents());

        if ($respuesta->estatus) {
            foreach ($respuesta->notificaciones as $n) {
                $datos = Json_decode(json_encode($n), true);

                $fecha = Carbon::parse($datos['created_at']);
                $datos['fecha'] = $fecha->format('d/m/Y');

                $notificacion = new Notificacion($datos);
                $notificaciones[] = $notificacion;
            }
        }
        //dd($notificaciones);
        return $notificaciones;
    }

    public function getGrupos() {
        $grupos = [];
        $peticion = Http::get('http://localhost:3000/grupos/');
        $respuesta = json_decode($peticion->getBody()->getContents());

        if($respuesta->estatus) {
            foreach($respuesta->grupos as $g) {
                $grupo = [$g->id => $g->nombre];
                $grupos[] = $grupo;
            }
        }

        return $grupos;
    }

    public function enviar(Request $request) {
        $this->validate($request, [
            'titulo' => 'required',
            'descripcion' => 'required'
        ]);

        $sendData = [
            'titulo' => $request->titulo,
            'descripcion' => $request->descripcion,
            'grupo_id' => $request->grupo
        ];

        $peticion = Http::post('http://localhost:3000/notificaciones/', $sendData);
        $respuesta = json_decode($peticion);

        if($respuesta->estatus) {
            return redirect(route('notificaciones'));
        } 

        return Redirect::back()->withErrors([$respuesta->mensaje]);
    }
}
